_adr/87/index.adoc

```diff
@@ -1,48 +1,451 @@
+
+KAS Fleet Manager controls the labels exposed for each metric returned in all endpoints. 
+Metrics retrieved from Observatorium are also filtered so that end users will only receive metrics for their own Kafka instance.
+Apart from these, no additional aggregation, filtering or modification of the metric is done by KAS Fleet Manager.
 
 ## Proposal
-// What is the decision being proposed
+### Metric and Label Naming
+All metric and label names must adhere to the policy stated in link:../../_ap/16/index.adoc[AP-16: Naming Policy for Customer Facing Metrics].
+
+### Kafka Metric Categories
+The customer facing Kafka metrics can be divided into multiple categories.
+Categorizing these metrics into logical groups can help us formulate the metric name and document it.
+
+|===
+|Category | Description | Sub-component (to be specified in the metric name)
+
+|Instance
+|General Kafka metrics not specific to any other sub-components.
+|instance
+
+|Broker
+|Metrics associated to Kafka brokers
+|broker
+
+|Topic
+|Metrics associated to Kafka topics
+|topic (Even though the topic metrics already have a 'topic' label, including this improves the clarity of the metric name)
+|===
+
+Examples of sub-component usage in the metric name:
+
+* For *Instance* metrics, the metric prefix would be `kas_instance_`.
+* For *Broker* metrics, the metric prefix would be `kas_broker_`.
+* For *Topic* metrics, the metric prefix would be `kas_topic_`.
+
+### Converted Metrics
+At minimum, each metric should have a label that identifies the Kafka instance if possible.
+This would be useful if a user has multiple Kafka instances.
+If they want to integrate all of their Kafka metrics into one monitoring system, the label(s) can act as an identifier.
+For Kafka metrics, we can use `instance_name` and `instance_id` to identify the Kafka instance the metric belongs to.
+
+#### Cluster Metrics
+|===
+|Old Metric Name | Revised Metric Name | Type | Old Labels | Revised Labels | Description
+
+|kafka_instance_max_message_size_limit
+|kas_instance_message_size_limit_bytes
+|Gauge
+|Not available
+|instance_name, instance_id
+|Maximum message size, in bytes, supported by the instance.
+
+|kafka_instance_partition_limit
+|kas_instance_partition_limit
+|Gauge
+|Not available
+|instance_name, instance_id
+|Maximum number of partitions per instance
+
+|kafka_namespace:haproxy_server_bytes_in_total:rate5m
+|kas_instance_incoming_bytes_rate5m
```

Comment:
I questioning the final underscore separator in metrics that convey rates.  By convention Prometheus uses `:rate_xt_`

in the `_rate5m` suffix. By convention the las    Prometheus's convention is colon.   

kas_instance_outgoing_bytes_total
kas_instance_outgoing_bytes:rate5m

Reply:
In Prometheus convention, the `:` are reserved for user defined recording rules. It may be often used for rates because they would usually be used for these. Do we want to mix this in our label name? We will have the following types then:

* kas_\<metric\>_\<unit: e.g. bytes|seconds\>
* kas_\<metric\>_\<type/aggregate: e.g. total|info|bucket\>
* kas_\<metric\>:ratent

The suffix may not always be a unit either, it could be a metric type or aggregate i.e. `total` or `info`. Should this rule apply to those as well?

Reply:
Thanks for the feedback. On reflection, let's go with your idea.